app/src/main/java/org/thunderdog/challegram/data/TGMessage.java

```diff
@@ -3253,8 +3253,16 @@ protected final boolean moveBubbleTimePartToLeft () {
     return Lang.rtl();
   }
 
-  protected final boolean needExpandBubble (int bottomLineContentWidth) {
-    return bottomLineContentWidth + bubbleTimePartWidth > (allowBubbleHorizontalExtend() ? pRealContentMaxWidth : Math.min(pRealContentMaxWidth, bubbleInnerWidth));
+  protected final boolean needExpandBubble (int bottomLineContentWidth, int bubbleTimePartWidth, int maxLineWidth) {
+    switch (bottomLineContentWidth) {
+      case BOTTOM_LINE_KEEP_WIDTH:
+        return false;
+      case BOTTOM_LINE_EXPAND_HEIGHT:
+        return true;
+      case BOTTOM_LINE_DEFINE_BY_FACTOR:
+        throw new UnsupportedOperationException();
```

Comment:
I was worried that this might trigger some edge-case crashes, but it never happened on beta with 490k+ users, so it seems to be OK to merge.